File: kernel/kernel_union/src/main/java/com/ycr/kernel/union/helper/JsonHelper.java
package com.ycr.kernel.union.helper;

import android.content.Context;

import com.ycr.kernel.json.parse.IJsonElement;
import com.ycr.kernel.json.parse.IJsonParser;

import org.jetbrains.annotations.Nullable;

import java.lang.reflect.Type;

/**
 * created by yuchengren on 2019/4/22
 */
public class JsonHelper{

    private static IJsonParser jsonParser = UnionContainer.INSTANCE.getJsonParser();

    public static String toJson(@Nullable Object any) {
        return jsonParser.toJson(any);
    }

    public static <T> T fromJson(@Nullable String json, @Nullable Type type) {
        return jsonParser.fromJson(json,type);
    }

    public static <T> T fromJson(@Nullable IJsonElement json, @Nullable Type type) {
        return jsonParser.fromJson(json,type);
    }

    public static IJsonElement parseJsonToElement(@Nullable String json) {
        return jsonParser.parseJsonToElement(json);
    }
}

File: kernel/kernel_union/src/main/java/com/ycr/kernel/union/helper/JsonHelper.kt
package com.ycr.kernel.union.helper

import android.content.Context

import com.ycr.kernel.json.parse.IJsonElement
import com.ycr.kernel.json.parse.IJsonParser

import java.lang.reflect.Type

/**
 * created by yuchengren on 2019/4/22
 */
object JsonHelper {

    private val jsonParser = UnionContainer.jsonParser

    fun toJson(any: Any?): String? {
        return jsonParser.toJson(any)
    }

    fun <T> fromJson(json: String?, type: Type?): T? {
        return jsonParser.fromJson<T>(json, type)
    }

    fun <T> fromJson(json: IJsonElement?, type: Type?): T? {
        return jsonParser.fromJson<T>(json, type)
    }

    fun parseJsonToElement(json: String?): IJsonElement? {
        return jsonParser.parseJsonToElement(json)
    }
}
